### PR TITLE
Get secrets in settings

### PR DESCRIPTION
**Pull request by jolheiser**
This PR adds the ability to get a `setting` from a `secret`. 
I've opted to follow the format Drone does to ease migration for users, let me know if we want to go another route.

```yaml
settings:
  my_secret:
    from_secret: secret_name
```

-----

I'm not a huge fan of the hadoken logic, but with our current YAML unmarshaling I think it's the best we can do.

### CODE REVIEW

File: pipeline/frontend/yaml/compiler/params.go
Comment (6543) [inline pipeline/frontend/yaml/compiler/params.go:69]: if secrets != nil

File: pipeline/frontend/yaml/compiler/params_test.go
Comment (6543) [inline pipeline/frontend/yaml/compiler/params_test.go:62]: can you add an error case too?

File: pipeline/frontend/yaml/compiler/params.go
Comment (jolheiser) [reply pipeline/frontend/yaml/compiler/params.go:69]: Since we are only reading we don't need a nil check. Nil maps behave like empty maps when reading,

File: pipeline/frontend/yaml/compiler/params_test.go
Comment (jolheiser) [reply pipeline/frontend/yaml/compiler/params_test.go:61]: Added below separately

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@jolheiser one thing i just remember - can you add some docu for this feature :laughing:

**Comment by jolheiser** [comment]
> @jolheiser one thing i just remember (before hit the merge button ;) ) - can you add some docu for this feature laughing

Good catch! I've added some docs, will wait for review before merging.

**Review by 6543 — APPROVED** [review]
not to short and not to long :)

**Pull request merged**
